centreon/src/Core/GraphTemplate/Domain/Model/GraphTemplate.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\GraphTemplate\Domain\Model;
+
+use Assert\AssertionFailedException;
+use Centreon\Domain\Common\Assertion\Assertion;
+
+class GraphTemplate
+{
+    public const NAME_MAX_LENGTH = 200;
+    public const LABEL_MAX_LENGTH = 200;
+    public const BASE_ALLOWED_VALUES = [1000, 1024];
+
+    /**
+     * @param int $id
+     * @param string $name
+     * @param string $verticalAxisLabel
+     * @param int $width
+     * @param int $height
+     * @param int $base
+     * @param null|float $gridLowerLimit
+     * @param null|float $gridUpperLimit
+     * @param bool $isUpperLimitSizedToMax
+     * @param bool $isGraphScaled
+     * @param bool $isDefaultCentreonTemplate
+     *
+     * @throws AssertionFailedException
+     */
+    public function __construct(
+        private readonly int $id,
+        private string $name,
+        private string $verticalAxisLabel,
+        private int $width,
+        private int $height,
+        private int $base = 1000,
+        private ?float $gridLowerLimit = null,
+        private ?float $gridUpperLimit = null,
+        private bool $isUpperLimitSizedToMax = false,
+        private bool $isGraphScaled = false,
+        private bool $isDefaultCentreonTemplate = false,
+    ) {
+        Assertion::positiveInt($id, 'GraphTemplate::id');
+
+        Assertion::notEmptyString($name, 'GraphTemplate::name');
+        Assertion::maxLength($name, self::NAME_MAX_LENGTH, 'GraphTemplate::name');
+
+        Assertion::notEmptyString($verticalAxisLabel, 'GraphTemplate::verticalAxisLabel');
+        Assertion::maxLength($verticalAxisLabel, self::LABEL_MAX_LENGTH, 'GraphTemplate::verticalAxisLabel');
+
+        Assertion::inArray($base, self::BASE_ALLOWED_VALUES, 'GraphTemplate::base');
+
+        if ($this->isUpperLimitSizedToMax === true) {
```

Comment:
This is a hard typed boolean, strictness with `true` is unnecessary.
```suggestion
        if ($this->isUpperLimitSizedToMax) {
```

Reply:
Addendum : Is it a domain failure (inconsistency) if we set `gridUpperLimit` to `true|false` while having `isUpperLimitSizedToMax` set to `true` ?
- if yes, check the case and throw an assert exception
- if no, please add a comment which tells explicitely that we ignore such cases (quietly)